src/game.cpp

```diff
@@ -316,6 +316,85 @@ static inline u8 getNeighbors(v3s16 p, INodeDefManager *nodedef, ClientMap *map,
 	return neighbors;
 }
 
+namespace {
+class Interpolator
+{
+public:
+	Interpolator(v3f start, v3f end):
+		m_start(start),
+		m_end(end),
+		m_current(start),
+		m_done(false)
+	{
+		m_step_amount = end - start;
+
+		double largest_increase_amount = fabs(m_step_amount.X);
+
+		if (fabs(m_step_amount.Y) > largest_increase_amount)
+		{
+			largest_increase_amount = fabs(m_step_amount.Y);
+		}
+
+		if (fabs(m_step_amount.Z) > largest_increase_amount)
+		{
+			largest_increase_amount = fabs(m_step_amount.Z);
+		}
+
+		if (largest_increase_amount == 0)
+		{
+			m_done = true;
+			return;
+		}
+
+		m_step_amount /= largest_increase_amount;
+	}
+
+	void Step(std::vector<v3s16>& points)

```

Comment:
Never ever do non-const references. They are forbidden by the code style and are generally a very bad idea.